src/sk_transformers/generic_transformer.py

```diff
@@ -192,16 +191,44 @@ class AggregateTransformer(BaseTransformer):
     ```
 
     Args:
-        features (List[Tuple[str, str, List[str]]]): List of tuples containing the column identifiers and the aggregation function(s).
-                The first column identifier (features[0]) is the column that will be used to group the data.
-                It can be either numerical or categorical. The second column identifier (features[1]) is the column that will be used
-                for aggregations. This column must be numerical.
+        features (List[Tuple[str, Tuple[str, Any, str]]]): List of tuples where
+            the first element is the name or the list of names of columns to be grouped-by,
+            and the second element is a tuple or list of tuples containing the aggregation information.
+            In this tuple, the first element is the name of the column to be aggregated,
+            the second element is the aggregation function as a string or function object,
+            and the third element is the name of the new aggregated column.
     """
 
-    def __init__(self, features: List[Tuple[str, str, List[str]]]) -> None:
+    def __init__(
+        self,
+        features: List[
+            Tuple[
+                Union[str, List[str]],
+                Union[Tuple[str, Any, str], List[Tuple[str, Any, str]]],
+            ]
+        ],
+    ) -> None:
         super().__init__()
         self.features = features
 
+    def __get_list_of_features(self) -> List:
+        feature_list = []
+
+        for groupby_columns, agg_features in self.features:
+            if isinstance(groupby_columns, str):
+                feature_list.append(groupby_columns)
+            else:
+                for groupby_column in groupby_columns:
+                    feature_list.append(groupby_column)
+
+            if isinstance(agg_features, tuple):
+                feature_list.append(agg_features[0])
+            else:
```

Comment:
In this level of complexity, some more data type checks could help to prevent hard-to-find errors. Maybe not only check for `tuple` but also for `list` and raise an error if it is something different. What do you think?